processfamily/__init__.py

```diff
@@ -381,9 +366,155 @@ def _handle_response_line(self, line):
 CPU_AFFINITY_STRATEGY_CHILDREN_ONLY = 1
 CPU_AFFINITY_STRATEGY_PARENT_INCLUDED = 2
 
-CHILD_COMMS_STRATEGY_NONE = 0
-CHILD_COMMS_STRATEGY_PIPES_CLOSE = 1
-CHILD_COMMS_STRATEGY_PROCESSFAMILY_RPC_PROTOCOL = 2
+
+class NoCommsStrategy(ChildCommsStrategy):
+    MONITOR_STDOUT = False
+    CAN_WAIT_FOR_TERMINATE = False
+
+    @staticmethod
+    def get_popen_streams(echo_std_err):
+        """Returns kwargs for stdin, stdout and stderr to pass to subprocess.Popen"""
+        return {"stdin": None, "stdout": None, "stderr": None}
+
+class ClosePipesCommsStrategy(ChildCommsStrategy):
+    def stop_child(self, end_time):
+        p = self._process_instance
+        logger.info("Closing stdin for process %r", self)
+        try:
+            p.stdin.close()
+        except Exception as e:
+            logger.warning("Failed to close child process input stream with PID %s: %s\n%s", self.pid, e, _traceback_str())
+        yield
+        yield
```

Comment:
If this is only meant to close after cleanup, perhaps we should wait for the process to end here?  Or do we wait somewhere else?  I suppose what I'm wondering is, do we have a backup plan if this process misbehaves and doesn't shut down when stdin is closed?